### PR TITLE
Use event_ids array in GetEventLogsWithStatusByEnvName

### PR DESCRIPTION
**Pull request by wwalexander**
I updated GetEventLogsByEnvName to use the new schema, but didn't see GetEventLogsWithStatusByEnvName, which just adds an extra column to the query and is what is actually called by the API. This PR makes the two methods use the same query structure.

### CODE REVIEW

**Review by cleanshavenalex — APPROVED** [review]
lgtm

**Review by bkeroackdsc — APPROVED** [review]

**Comment by wwalexander** [comment]
I've added a duplicate check when adding event logs to an environment after realizing a temporary issue where I was inserting a duplicate event. The issue is fixed now, but I figure the check is worth keeping the DB in a sane state if an event is for some reason repeatedly associated with an environment.

File: pkg/persistence/datalayer_test.go [hunk 1]
```diff
@@ -1213,6 +1214,11 @@ func TestDataLayerGetEventLogsByEnvName(t *testing.T) {
 		t.Fatalf("should have succeeded: %v", err)
 	}
 	if i := len(logs); i != 2 {
+		for _, l := range logs {
+			log.Println(l.ID)
+			log.Println(l.Created)
+		}
+		t.Errorf("%v", logs) // DELETEME
```
Comment (bkeroackdsc) [inline pkg/persistence/datalayer_test.go:1221]: `DELETEME`

**Review by bkeroackdsc — APPROVED** [review]

File: pkg/persistence/datalayer_test.go (hunk 1 above)
Comment (wwalexander) [reply pkg/persistence/datalayer_test.go]: Ah, apologies! Fixed by 6c4a72d.

**Pull request merged**